src/redis-benchmark.c

```diff
@@ -361,9 +361,10 @@ static redisConfig *getRedisConfig(const char *ip, int port,
     if (hostsocket == NULL) fprintf(stderr, "%s:%d\n", ip, port);
     else fprintf(stderr, "%s\n", hostsocket);
     int abort_test = 0;
-    if (!strncmp(reply->str,"NOAUTH",5) ||
-        !strncmp(reply->str,"WRONGPASS",9) ||
-        !strncmp(reply->str,"NOPERM",5))
+    if (reply->type == REDIS_REPLY_ERROR &&
```

Comment:
Looking at the code it seems like `reply` might be NULL. So this fix still doesn't avoid a segfault.